lib/LaTeXML/Package/pst-node.sty.ltxml

```diff
@@ -180,39 +180,46 @@ DefMacro('\Rnode PSCoord', sub {
     (T_CS('\@Rnode'), T_OTHER('('), Explode($coord), T_OTHER(')')); });
 
 DefConstructor('\@Rnode PSCoord {} {}', '#3',
-  alias => '\Rnode',
+  alias       => '\Rnode',
   afterDigest => sub { storeNode($_[1]->getArg(2), $_[1]->getArg(1)); });
 
 DefConstructor('\pnode ZeroPSCoord {}', '',
   afterDigest => sub { storeNode($_[1]->getArg(2), $_[1]->getArg(1)); });
 
 DefPSConstructor('\cnode', 'ZeroPSCoord {PSDimension} {}',
-  "<ltx:circle " . $PSLineAttr . "%&PairAttr(#2,x,y) r='&ptValue(#3)' fill='#fill' />",
-  attributes => [qw(fill linecolor linewidth dash)],
-  afterDigest => sub { storeNode($_[1]->getArg(4), $_[1]->getArg(2)); });
+  "<ltx:circle " . $PSLineAttr . "x='#x' y='#y' r='&pxValue(#3)' fill='#fill' />",
+  attributes  => [qw(fill linecolor linewidth dash)],
+  afterDigest => sub { storeNode($_[1]->getArg(4), $_[1]->getArg(2)); },
+  properties  => sub {
+    (x => $_[2]->getX->pxValue, y => $_[2]->getY->pxValue); });
 
 DefPSConstructor('\Cnode', 'ZeroPSCoord {}',
-  "<ltx:circle " . $PSLineAttr . "%&PairAttr(#2,x,y) r='&ptValue(#radius)' fill='#fill' />",
-  attributes => [qw(fill linecolor linewidth dash radius)],
-  afterDigest => sub { storeNode($_[1]->getArg(4), $_[1]->getArg(2)); });
+  "<ltx:circle " . $PSLineAttr . "x='#x' y='#y' r='&pxValue(#radius)' fill='#fill' />",
+  attributes  => [qw(fill linecolor linewidth dash radius)],
+  afterDigest => sub { storeNode($_[1]->getArg(4), $_[1]->getArg(2)); },
```

Comment:
Well here I am counting 2 arguments in the pattern, but again `getArg(4)`. This one has to be a typo? Maybe it should be 3?

Ah no! There is also an `OptionalMatch:*` prepended to the arguments, so the total number is +2 the specified ones. Definitely a learning curve for me here.

I'm starting to wonder whether a simplifying refactor would be for these prepended arguments of `DefPSConstructor` to get stored under the named properties with some invented names (e.g. "#isstar" and "#arrow"), in order to not have to always carry a +2 in one's mind when using the macro-specific arguments.

Reply:
Yep, `DefPSConstructor` is very unpleasant;  I'm putting of refactoring till I deal with going direct-to-svg (like pgf). Also, I had a hard time finding sensible test cases that still work w/LaTeX (you at least have to go old-skool `latex+dvi+xdvi` or similar.)